input/drivers/android_input.c

```diff
@@ -869,7 +871,7 @@ static void handle_hotplug(android_input_data_t *android_data,
    if (!back_mapped && settings->input.back_as_menu_toggle_enable)
       settings->input.autoconf_binds[*port][RARCH_MENU_TOGGLE].joykey = AKEYCODE_BACK;
 
-   android_data->pad_states[android_data->pads_connected].id = id;
+   android_data->pad_states[*port].id = id;
```

Comment:
Yes, I agree that it is not easy to work right now :)

From my testing I found that when I handle it like in the pull request it will not overwrite the controller in port 1 when I swap the controllers in port 0.

The current code is working fine for the NVIDIA button on the controller. But the way it is handled prevents you from using other Shield TV buttons, such as the ones on a CEC remote (Not the SHIELD remote). That's the advantage you get when mapping it to a separate controller.

About the potential OSD spam, this will be very minor on the Shield TV. It is only when you switch between using a remote or controller that it will let you know that you've switched or when you press the NVIDIA button. I would imagine, that there would be much more spam with other controllers that have two HID devices where the gameplay buttons are divided between the devices, but on the Shield TV that is not the case.